charts/opentelemetry-operator/templates/role.yaml

```diff
@@ -1,4 +1,5 @@
 {{- if .Values.role.create }}
+{{- if .Values.manager.leaderElection.enabled }}
```

Comment:
@TylerHelmuth that was my initial idea, but kept it like this for future roles that are not related to leaderElection
Otherwise 1 leaderElection flag would be sufficient to address this . wdyt ?

Reply:
@TylerHelmuth I think you are right. We can later add 2 or more conditions for future roles. Updated